scipy/interpolate/interpolate.py

```diff
@@ -221,8 +221,8 @@ def __init__(self, x, y, z, kind='linear', copy=True, bounds_error=False,
             kx = ky = {'linear': 1,
                        'cubic': 3,
                        'quintic': 5}[kind]
-        except KeyError:
-            raise ValueError("Unsupported interpolation type.")
+        except KeyError as e:
```

Comment:
since you're looking at these changes anyway could we make the changes be in favor of improving the information content of the error messages? 

for example in this one, rather than write: 
`ValueError("Unsupported interpolation type.")`
it could read 
`ValueError("{0} Unsupported interpolation type {1}.".format(key, type(key))` ?

The benefit is that someone reading error message (or logs) will determine that `key` has type `X` (whatever) and that is not supported for interpolation. Whereas as this PR reads, user only learns that there is a type w/unsupported interpolation (but does not know what that unsupported type is). 

Thoughts?

Reply:
That sounds like a good direction, but since this is a change in many files across the repo, I think we should be extra conservative. Any improvements should be made on separate PRs that are either module-specific or specific to the type of change being done.

Reply:
> That sounds like a good direction, but since this is a change in many files across the repo, I think we should be extra conservative. Any improvements should be made on separate PRs that are either module-specific or specific to the type of change being done.

Fair enough, perhaps you are interested to open (some of) those PR(s)?

Reply:
Yes. Would you be okay with f-strings instead of `.format`? Since SciPy requires Python 3.6 I think these are okay now. Also, I assume you meant after this PR is merged, because otherwise we'll have a merge conflict.

Reply:
> Yes. Would you be okay with f-strings instead of `.format`? Since SciPy requires Python 3.6 I think these are okay now. 

I'm not sure what the preference is on f-string versus, `.format` the scipy code base has both "old style" interpolation w/ `%` as well as the `.format`. I'm not aware of any use of f-strings but my github searching skills are not at the level where a query on `f"` or `f\"` will return anything for the repo. I'll ask on the dev-list what the preference is for the string interpolation and post here once reply. 

>Also, I assume you meant after this PR is merged, because otherwise we'll have a merge conflict.

Either way is fine, merge conflicts can be resolved on a rebase or you could wait until this one is merged.

Reply:
I understand. I'll wait until we're done with fixing the exception causes, then open the separate PR.

Reply:
Agree with this conversation: separate PR for follow-ups, and using f-strings is fine for new code or code you're touching anyway.